content/projects/oop-for-data-science/_index.md

```diff
@@ -1,3 +1,37 @@
 ---
 title: OOP for data science
+ready: False
 ---
+
+## Prerequisite
+
+You need to complete the Multivariate Regression assignment using the salary dataset before doing this assignment ({{% contentlink "projects/data-science-specific/multivariate-linear-regression" %}}). You will use the data and the model you built in that assignment for this OOP assignment.
+
+## Assignment
+It is efficient to put machine learning models and other data science techniques into classes so that we can reuse them later and change attributes without changing the code behind these models. Independent concepts can also be put into independent classes: for example, the functioning of a cross-validate class should not affect the functioning of a linear regression class. These concepts are also referred to as high cohesion and low coupling.
+
+1. Create a class called `Metrics` that has methods to compute the Mean Squared Error, Root Mean Squared Error, R-squared and Adjusted R-squared. Name these methods `mse`, `rmse`, `r_squared` and `adj_r_squared`. The class should have the following parameters:
+
+- `data_dict`: dictionary containing x_train, x_test, y_train and y_test data
+- `model type`: an sklearn estimator instance (e.g. LinearRegression())
+
+2. Create a  class called `DiagnosticPlots` that has methods to plot the (1) residuals versus predicted values and (2) predicted versus observed values. These methods should be called (1) `residual_vs_predicted` and (2) `predicted_vs_observed`.
```

Comment:
`DiagnosticPlotter` ?

And does this one also take in a data_dict and a model_type?